File: XamarinBinding/BindingTest/BindingTest/ViewController.cs
using System;
using System.Diagnostics;
using UIKit;
using Foundation;
using PayGuardian;

namespace BindingTest
{
    public partial class ViewController : UIViewController
    {
		// If these 2 variables were commented no error would occur, else MT5216 occurs
		BPNPaymentRequest request;
		PayGuardianTransaction transaction;

        protected ViewController(IntPtr handle) : base(handle)
        {
            // Note: this .ctor should not contain any initialization logic.
        }

        public override void ViewDidLoad()
        {
            base.ViewDidLoad();
			// Perform any additional setup after loading the view, typically from a nib.

			// Just to see if ViewDidLoad is called
			View.BackgroundColor = UIColor.Blue;

			// FIXME: If the code below is uncommented,
			// an MT0000 error occurs along with a null reference exception description

			//request = new BPNPaymentRequest("1234", null, new NSDecimalNumber("5.00"),
			//                                new NSDecimalNumber("1.00"), null,
			//                               "TENDER_TYPE_CREDIT", "TRANSACTION_TYPE_SALE",
			//                               "paulpgtest", "57!sE@3Fm", @"682000",
			//                               "682001", null, null, null, "TERMINAL_TYPE_INGENICO_BLUETOOTH",
			//                                "TRANSACTION_INDUSTRY_TYPE_RESTAURANT", false, true);
			//transaction = new PayGuardianTransaction(request);
			//transaction.RunOnCompletion((payment, error) =>
			//{

			//}, (state) => 
			//{

			//});
		}

        public override void DidReceiveMemoryWarning()
        {
            base.DidReceiveMemoryWarning();
            // Release any cached data, images, etc that aren't in use.
        }
    }
}

File: XcodeDemo/payguardiantest/payguardiantest/ViewController.m
#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

@synthesize _transaction;
@synthesize _request;

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    self.view.backgroundColor = [UIColor blueColor];
    
    _request = [[BPNPaymentRequest alloc] initInvoiceNumber:@"1234" pnRefNum:nil amount:[NSDecimalNumber decimalNumberWithString:@"5.00"] tipAmount:[NSDecimalNumber decimalNumberWithString:@"1.00"] cashBackAmount:nil tenderType:TENDER_TYPE_CREDIT transactionType:TRANSACTION_TYPE_SALE username:@"paulpgtest" password:@"57!sE@3Fm" merchantCode:@"682000" merchantAccountCode:@"682001" paymentAccountNumber:nil token:nil expirationDate:nil terminalType:TERMINAL_TYPE_INGENICO_BLUETOOTH industryType:TRANSACTION_INDUSTRY_TYPE_RESTAURANT disableEmv:NO testMode:YES];
    _transaction = [[PayGuardianTransaction alloc] initWithPaymentRequest:_request];
    
    [_transaction runOnCompletion:^(BPNPayment* payment, NSError* error) {
        //Run on completion
    } onStateChanged:^(PayGuardianTransactionState state) {
        //State changed
    }];
}

@end
